apps/google-analytics-4/frontend/src/components/config-screen/assign-content-type/AssignContentTypeCard.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Box,
+  Button,
+  Card,
+  FormControl,
+  Select,
+  Stack,
+  TextInput,
+  TextLink,
+} from '@contentful/f36-components';
+import { css } from 'emotion';
+import { AllContentTypes, ContentTypeEntries } from 'types';
+
+interface Props {
+  allContentTypes: AllContentTypes;
+  contentTypeEntries: ContentTypeEntries;
+  onContentTypeChange: (prevKey: string, newKey: string) => void;
+  onContentTypeFieldChange: (key: string, field: string, value: string) => void;
+  onAddContentType: () => void;
+  onRemoveContentType: (key: string) => void;
+}
+
+const styles = {
+  contentTypeItem: css({
+    flex: 4,
+  }),
+  removeItem: css({
+    flex: 1,
+  }),
+};
+
+const AssignContentTypeCard = (props: Props) => {
+  const {
+    allContentTypes,
+    contentTypeEntries,
+    onContentTypeChange,
+    onContentTypeFieldChange,
+    onAddContentType,
+    onRemoveContentType,
+  } = props;
+
+  return (
+    <Card>
+      {Object.keys(contentTypeEntries).length ? (
+        <Stack marginBottom="none" spacing="spacingXs">
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>Content type</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>Slug field</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.contentTypeItem}>
+            <FormControl marginBottom="none">
+              <FormControl.Label>URL prefix</FormControl.Label>
+            </FormControl>
+          </Box>
+          <Box className={styles.removeItem}></Box>
+        </Stack>
+      ) : null}
+      {Object.entries(contentTypeEntries).map(([key, { slugField, urlPrefix }], index) => {
```

Comment:
Second comment is all these `Object.entries/Object.keys`. 

I think the JSX would be much easier if we had some state, and then have a useEffect that set these states with the types of arrays you want.

That way the JSX can read something more like 
```
{some useful variable name}.map(some blah blah JSX)
```
and then you know by looking at the JSX immediately what you are looping over.

Reply:
This is a great suggestion; I have made this update